### PR TITLE
Document stories and questions around homepage popular links

### PR DESCRIPTION
**Pull request by syed-ali-tw**
Document stories and questions around homepage popular links

As we moved editing and publishing of 'Popular on GOV.UK'  to the Mainstream Publisher, we made certain design and implementation decisions which we have tried to capture through this PR.

[Trello](https://trello.com/c/QwQuxFIC/303-document-popular-links-schema-in-devdocs)
---
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: docs/homepage.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#### Which schema in publishing api does popular links use?
+
+link_collection
+
+#### Why popular links as a separate content item instead of moving the whole of homepage to publisher?
+
+Currently our use case is only to allow publishing of popular links and not any other content on the homepage, though in the future we might want to allow other content on the homepage to be published via Mainstream Publisher. Implementation-wise, we have investigated and understood how to allow publishing of popular link as a separate content item and link it to the homepage, and we think it is a simple approach to take at this point in time. If we want to allow other content items on the homepage to be publishable we can still create each of them as a separate content item and publish them separately, or move to a strategy of publishing homepage as a single content item. Though this strategy will require us to investigate more on how that can be done and understand the complexity and feasibility of it.
+
+#### Why is saving and publishing popular links is a synchronous operation instead of asynchronous (via sidekiq)?
```
Comment (mtaylorgds) [inline docs/homepage.md:34]: > Why is saving and publishing popular links is a synchronous operation instead of asynchronous (via sidekiq)?
Having re-read this, maybe this should be "Why is saving and publishing popular links a synchronous instead of asynchronous (via sidekiq)?"

File: docs/homepage.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+There are few reasons why we chose to keep it synchronous:
```
Comment (mtaylorgds) [inline docs/homepage.md:36]: > There are few reasons
There are **a** few reasons

**Review by mtaylorgds — COMMENTED** [review]
A couple of minor grammatical comments.

File: docs/homepage.md (hunk 1 above)
Comment (mtaylorgds) [reply docs/homepage.md]: I'm so sorry, I've just realised my suggestion was also not grammatically correct. It should be "Why is saving and publishing popular links synchronous instead of asynchronous (via sidekiq)?" (remove "a").

**Review by mtaylorgds — APPROVED** [review]

File: docs/homepage.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Homepage
+
+## Popular Links
+
+Popular links are the links that are present on GOV.UK under _'Popular on GOV.UK'_.
+These links are maintained through this Mainstream Publisher app.
+
+### How to edit popular links
+
+Popular links can be accessed and edited via the 'Homepage' link on Signon, for this we need the 'homepage_editor' permission inside the Mainstream Publisher app.
```
Comment (ChrisBAshton) [inline docs/homepage.md:10]: Drive-by comment (found this PR via https://github.com/alphagov/govuk-developer-docs/pull/4763): why was the decision made to expose the 'homepage' functionality of Mainstream Publisher in Signon itself (instead of relying on users to navigate to the functionality via clicking "Publisher")?

It seems a bit of a confusing gotcha to add, unless I'm missing some clear rationale :/

Comment (mtaylorgds) [reply docs/homepage.md:10]: @ChrisBAshton This is a product decision about trialing the Signon links as more task-based than app-based. You can chat with our product owner (Emma Carter) if you'd like to learn more.

Comment (ChrisBAshton) [reply docs/homepage.md:10]: Ah, thanks for the context - I'll ask Emma 👍 in the meantime it may be useful to add "Why is this being added directly to the Signon landing page?" to the "Questions" section in this doc.

Comment (syed-ali-tw) [reply docs/homepage.md:10]: Hey @ChrisBAshton , we have now updated the doc to add this question.

Comment (ChrisBAshton) [reply docs/homepage.md:10]: Perfect - thanks @syed-ali-tw
